### PR TITLE
Improvements to API/module docs

### LINKED ISSUE #21: Outdated version referenced in site docs
Hi, Mux! First, thanks so much for building a first-class Elixir package! ❤️ 

The version listed in the [Elixir SDK docs](https://mux.com/for/elixir/) could use an update, though :)

I wasn't sure the best place to call this out, so I hope you don't mind my doing so here. Thanks, again!

```diff
def deps do
  [
-    {:mux, "~> 1.4.0"}
+    {:mux, "~> 1.8.0"}
  ]
end

### PR DESCRIPTION
**Pull request by mcrumm**
Since I can't ~fix~ work on #21, here are some docs improvements, instead :) What do you think?

mux_docs_update

### CODE REVIEW

File: lib/mux.ex [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
-  @moduledoc """
-  Core module for Mux. You can use the `client` method to create authenticated
-  [Tesla client](https://hexdocs.pm/tesla/Tesla.html#build_client/2), which is passed to each
-  request from then on.
-  """
+  @external_resource "README.md"
+  @moduledoc @external_resource
+             |> File.read!()
+             |> String.split("<!-- MDOC !-->")
+             |> Enum.fetch!(1)
```
Comment (mmcc) [inline lib/mux.ex:6]: Clever!

**Comment by mmcc** [comment]
Thanks for the PR, @mcrumm!! I'd love to send you a care package as a thank you if you're interested :) I'll DM you on Twitter with details.

**Comment by mcrumm** [comment]
That is absolutely not necessary @mmcc... yes, of course, i'd love one :)

File: lib/mux.ex (hunk 1 above)
Comment (mcrumm) [reply lib/mux.ex:6]: José did this for `:phoenix_live_dashboard` and I was all 🤯

**Pull request merged**